packages-exp/app-check-exp/src/providers.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FirebaseApp } from '@firebase/app-exp';
+import { Provider } from '@firebase/component';
+import { exchangeToken, getExchangeRecaptchaTokenRequest } from './client';
+import { AppCheckError, ERROR_FACTORY } from './errors';
+import { AppCheckProvider, AppCheckToken } from './public-types';
+import { getToken as getReCAPTCHAToken } from './recaptcha';
+import { AppCheckTokenInternal } from './state';
+
+/**
+ * App Check provider that can obtain a reCAPTCHA V3 token and exchange it
+ * for an AppCheck token.
+ *
+ * @public
+ */
+export class ReCaptchaV3Provider implements AppCheckProvider {
+  /**
+   * @internal
+   */
+  _delegate: ReCaptchaV3ProviderImpl;
+  /**
+   * Create a ReCaptchaV3Provider instance.
+   * @param siteKey - ReCAPTCHA V3 siteKey.
+   */
+  constructor(siteKey: string) {
+    this._delegate = new ReCaptchaV3ProviderImpl(siteKey);
+  }
+  /**
+   * Returns an AppCheck token.
+   */
+  getToken(): Promise<AppCheckToken> {
+    return this._delegate.getToken();
+  }
+}
+
+export class ReCaptchaV3ProviderImpl implements AppCheckProvider {
```

Comment:
Can we just implement things in `ReCaptchaV3Provider` instead of using `_delegate` with another class?

Reply:
Done.